benchmarks/src/imdb/mod.rs

```diff
@@ -0,0 +1,236 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Benchmark derived from IMDB dataset.
+
+use datafusion::{
+    arrow::datatypes::{DataType, Field, Schema},
+    common::plan_err,
+    error::Result,
+};
+mod convert;
+pub use convert::ConvertOpt;
+
+use std::fs;
+mod run;
+pub use run::RunOpt;
+
+// we have 21 tables in the IMDB dataset
+pub const IMDB_TABLES: &[&str] = &[
+    "aka_name",
+    "aka_title",
+    "cast_info",
+    "char_name",
+    "comp_cast_type",
+    "company_name",
+    "company_type",
+    "complete_cast",
+    "info_type",
+    "keyword",
+    "kind_type",
+    "link_type",
+    "movie_companies",
+    "movie_info_idx",
+    "movie_keyword",
+    "movie_link",
+    "name",
+    "role_type",
+    "title",
+    "movie_info",
+    "person_info",
+];
+
+/// Get the schema for the IMDB dataset tables
+/// see benchmarks/data/imdb/schematext.sql
+pub fn get_imdb_table_schema(table: &str) -> Schema {
+    match table {
+        "aka_name" => Schema::new(vec![
+            Field::new("id", DataType::Int32, false),
```

Comment:
Assuming that there are no negative ids, should be use `UInt32` instead of `Int32`?